### PR TITLE
Add `stabilizer` for $G$-sets with action on `Set`, `Vector` or `Tuple` objects

### PR DESCRIPTION
**Pull request by ThomasBreuer**
The stabilizer of a `Set` of points in the G-set is the setwise stabilizer. The stabilizer of a `Vector` or `Tuple` of points in the G-set is the pointwise stabilizer.

### CODE REVIEW

File: src/Groups/gsets.jl [hunk 1]
```diff
@@ -450,17 +450,28 @@ julia> map(length, orbs)
...
+    stabilizer(Omega::GSet{T,S}, omega::Set{S}; check::Bool = true) where {T,S}
+    stabilizer(Omega::GSet{T,S}, omega::Vector{S}; check::Bool = true) where {T,S}
+    stabilizer(Omega::GSet{T,S}, omega::Tuple{Vararg{S}}; check::Bool = true) where {T,S}
 
 Return the subgroup of `G = acting_group(Omega)` that fixes `omega`,
 together with the embedding of this subgroup into `G`.
+
+If `omega` is a `Set` or a `Vector` or a `Tuple` of points in `Omega`
+then `stabilizer` means the setwise or pointwise stabilizer, respectively,
+of the entries in `omega`.
```
Comment (lgoettgens) [inline src/Groups/gsets.jl]: I had to read this sentence multiple times to understand it. Could you maybe split the setwise and the pointwise part into two different sentences?

File: src/Groups/gsets.jl [hunk 2]
```diff
@@ -474,6 +485,33 @@ function stabilizer(Omega::GSet{T,S}, omega::S; check::Bool = true) where {T,S}
     return stabilizer(G, omega, gfun)
 end
 
+# support `stabilizer` under "derived" actions:
+# If the given point is a set, tuple, or vector of the element type of
+# the G-set then compute the setwise or pointwise stabilizer, respectively.
```
Comment (lgoettgens) [inline src/Groups/gsets.jl]: Here similarly. The wait you explain it in the PR description is better understandablr

File: src/Groups/gsets.jl [hunk 3]
```diff
@@ -474,6 +485,33 @@ function stabilizer(Omega::GSet{T,S}, omega::S; check::Bool = true) where {T,S}
...
+
+function stabilizer(Omega::GSet{T,S}, omega::Set{S}; check::Bool = true) where {T,S}
+    check && @req all(x -> x in Omega, omega) "omega must be a set of elements of Omega"
```
Comment (lgoettgens) [inline src/Groups/gsets.jl]:
```suggestion
    check && @req all(in(Omega), omega) "omega must be a set of elements of Omega"
```
Avoid an anonymous function

File: src/Groups/gsets.jl [hunk 4]
```diff
@@ -474,6 +485,33 @@ function stabilizer(Omega::GSet{T,S}, omega::S; check::Bool = true) where {T,S}
...
+    G = acting_group(Omega)
+    gfun = action_function(Omega)
+    derived_fun = function(x, g) return Set([gfun(y, g) for y in x]); end
```
Comment (lgoettgens) [inline src/Groups/gsets.jl]:
```suggestion
    derived_fun = function(x, g) return Set(gfun(y, g) for y in x); end
```
I am not 100% sure this works. But if it does, it avoids allocating an intermediate vector

File: src/Groups/gsets.jl [hunk 5]
```diff
@@ -474,6 +485,33 @@ function stabilizer(Omega::GSet{T,S}, omega::S; check::Bool = true) where {T,S}
...
+    return stabilizer(G, omega, derived_fun)
+end
+
+function stabilizer(Omega::GSet{T,S}, omega::Vector{S}; check::Bool = true) where {T,S}
+    check && @req all(x -> x in Omega, omega) "omega must be a vector of elements of Omega"
```
Comment (lgoettgens) [inline src/Groups/gsets.jl]:
```suggestion
    check && @req all(in(Omega), omega) "omega must be a vector of elements of Omega"
```

File: src/Groups/gsets.jl [hunk 6]
```diff
@@ -474,6 +485,33 @@ function stabilizer(Omega::GSet{T,S}, omega::S; check::Bool = true) where {T,S}
...
+    G = acting_group(Omega)
+    gfun = action_function(Omega)
+    derived_fun = function(x, g) return [gfun(y, g) for y in x]; end
+    return stabilizer(G, omega, derived_fun)
+end
+
+function stabilizer(Omega::GSet{T,S}, omega::Tuple{Vararg{S}}; check::Bool = true) where {T,S}
+    check && @req all(x -> x in Omega, omega) "omega must be a tuple of elements of Omega"
```
Comment (lgoettgens) [inline src/Groups/gsets.jl]:
```suggestion
    check && @req all(in(Omega), omega) "omega must be a tuple of elements of Omega"
```

File: src/Groups/gsets.jl (hunk 1 above)
Comment (mjrodgers) [reply src/Groups/gsets.jl]: I actually feel like this is clear as written. Does it really need to be split up?

Comment (lgoettgens) [reply src/Groups/gsets.jl]: the ambiguity I see is that some of set, vector and tuple have setwise and some have pointwise stabilizers. But this sentence does not tell me which has which.
A sentence like in the PR description avoids this:
> The stabilizer of a Set of points in the G-set is the setwise stabilizer. The stabilizer of a Vector or Tuple of points in the G-set is the pointwise stabilizer.

**Review by mjrodgers — APPROVED** [review]
I think this is good as a first implementation, and all of the remarks of @lgoettgens seem to be addressed.

I think as a next step maybe we can implement these without computing the entire group action on sets/tuples; but I think we should merge this now and then I will open a separate issue for performance improvements.

**Pull request merged**